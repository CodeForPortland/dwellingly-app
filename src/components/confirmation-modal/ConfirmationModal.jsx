import React from 'react';
import PropTypes from 'prop-types';
import Card from '../card/Card';
// import { CARD_TYPES } from '../../constants/constants';
import Icon from '../icon/Icon';
import './ConfirmationModal.scss';

class ConfirmationModal extends React.Component {
  render() {
    // Render nothing if the "show" prop is false
    if(!this.props.show) {
      return null;
    }

    return (
      <Card className="confirmCard" >
        <Card.Content className="card__content">
          <Icon icon="checkbox"/>
          <p className="title">
            {this.props.children}
          </p>
          <div className="button-container">
            <button className="cancel" onClick={this.props.onClose}
              type="button">
              Cancel
            </button>
            <button className="btn" onClick={this.props.onClose}
              type="button">
              Confirm
            </button>
          </div>
        </Card.Content>
      </Card>
    );
  }
}

ConfirmationModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};

export default ConfirmationModal;
