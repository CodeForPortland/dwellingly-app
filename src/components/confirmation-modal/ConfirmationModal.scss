@import "../../constants/constants.scss";

.confirmCard {
  background: rgba(0, 0, 0, 0.6);
  justify-content: center;
  position: fixed;
  width: 100%;
  height: auto;

  .card__content {
    padding: 2.2em 2.5em;
    background: $color__white;
    width: 40%;
    border-radius: $border-radius--card;
    @include dialog_shadow;
    @include clearfix;
    align-self: center;
    text-align: center;

    .title {
      margin: 0;
      font-size: 1em;
    }
  }
}

.button-container {
  display: flex;
  justify-content: space-around;
  margin-top: 1.5em;

}

#yes-button {
  // margin-top: .5em;
  width: 8em;
}

.cancel {
  background: $color__gray--medium;
  border-radius: $border-radius--btn;
  color: $color__white;
  padding: 0 1rem;
  // margin-top: .5em;
  width: 8em;
}
