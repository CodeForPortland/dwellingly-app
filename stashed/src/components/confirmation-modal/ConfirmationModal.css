/* V A R I A B L E S */
/* end variables */
/* F U N C T I O N S */
/* end functions */
/* M I X I N S */
/* end mixins */
.full-window-modal {
  background: rgba(0, 0, 0, 0.6);
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  position: fixed;
  width: 100%;
  height: auto;
}

.full-window-modal .modal-content {
  padding: 2.2em 2.5em;
  background: #ffffff;
  width: 40%;
  border-radius: 10px;
  -webkit-box-shadow: 0 2px 3px rgba(0, 0, 0, 0.1);
          box-shadow: 0 2px 3px rgba(0, 0, 0, 0.1);
  -ms-flex-item-align: center;
      -ms-grid-row-align: center;
      align-self: center;
  text-align: center;
}

.full-window-modal .modal-content:after {
  content: "";
  display: table;
  height: 0;
  clear: both;
  visibility: hidden;
}

.full-window-modal .modal-content .title {
  margin: 0;
  font-size: 1em;
}

.button-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  margin-top: 1.5em;
}

#yes-button {
  width: 8em;
}

.cancel {
  background: #858585;
  border-radius: 20px;
  color: #ffffff;
  padding: 0 1rem;
  width: 8em;
}
/*# sourceMappingURL=ConfirmationModal.css.map */