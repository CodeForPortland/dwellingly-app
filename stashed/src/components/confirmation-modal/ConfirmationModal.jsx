import React from 'react';
import PropTypes from 'prop-types';
import './ConfirmationModal.scss';

class ConfirmationModal extends React.Component {
  render() {
    // Render nothing if the "show" prop is false
    if(!this.props.show) {
      return null;
    }

    return (
      <div className="full-window-modal" >
        <div className="modal-content">
          <p className="title">
            {this.props.children}
          </p>
          <div className="button-container">
            <button form='newPropertyForm' className="btn"
              id="yes-button" type="submit"
              onClick={this.props.onSubmit}>
              YES
            </button>
            <button className="cancel" onClick={this.props.onClose}
              type="button">
              NO
            </button>
          </div>
        </div>
      </div>
    );
  }
}

ConfirmationModal.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};

export default ConfirmationModal;
