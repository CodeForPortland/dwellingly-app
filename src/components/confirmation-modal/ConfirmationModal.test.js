import React from 'react';
import renderer from 'react-test-renderer';
import ConfirmationModal from './ConfirmationModal';

const ConfirmationModalTest = (props) =>
<ConfirmationModal
{...props}
/>;

describe('ConfirmationModal Component', () => {
  it('correctly renders ConfirmationModal component', () => {
    const ConfirmationModalComponent = renderer.create(ConfirmationModalTest).toJSON();
    expect(ConfirmationModalComponent).toMatchSnapshot();
  });

  it('renders component when show prop is true', () => {
    const props = {
      show: true,
    },
    ConfirmationModalComponent = mount(<ConfirmationModal {...props} />).find('p');
    expect(ConfirmationModalComponent.hasClass('title')).toEqual(true);
  });
});
