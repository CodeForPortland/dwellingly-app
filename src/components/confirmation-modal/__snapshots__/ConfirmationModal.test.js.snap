// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal Component correctly renders ConfirmationModal component 1`] = `null`;
